homeassistant/helpers/script.py

```diff
@@ -59,6 +56,30 @@
 ACTION_DEVICE_AUTOMATION = "device"
 ACTION_ACTIVATE_SCENE = "scene"
 
+IF_RUNNING_ERROR = "error"
+IF_RUNNING_IGNORE = "ignore"
+IF_RUNNING_PARALLEL = "parallel"
+IF_RUNNING_RESTART = "restart"
+# First choice is default
+IF_RUNNING_CHOICES = [
+    IF_RUNNING_PARALLEL,
+    IF_RUNNING_ERROR,
+    IF_RUNNING_IGNORE,
+    IF_RUNNING_RESTART,
+]
+
+RUN_MODE_BACKGROUND = "background"
+RUN_MODE_BLOCKING = "blocking"
+RUN_MODE_LEGACY = "legacy"
+# First choice is default
+RUN_MODE_CHOICES = [
+    RUN_MODE_BACKGROUND,
+    RUN_MODE_BLOCKING,
```

Comment:
Blocking should be the default.

Reply:
Remember, this is not just used by automations. It is also used by the script integration as well as a few others. The automation integration can define a different default for itself (via its config schema) and pass that in, effectively overriding the script helper's default. I chose background as the default specifically to try and be as compatible with the behavior of other integrations (besides automation & script) as possible. But, again, doesn't really matter to me.

Reply:
I think that we should pick the default that we want future integrations to use. I would expect that executing a script will take as long as the script takes to execute.

Like you said, we can change the default for script/automation to match legacy behavior for now.

Reply:
Ok, fair enough. I'll change the default to blocking.